app/controllers/user.js

```diff
@@ -87,3 +88,47 @@ exports.userList = async (req, res, next) => {
     next(err);
   }
 };
+
+exports.singUpAdmins = async (req, res, next) => {
+  const auth = req.headers[AUTHORIZATION];
+
+  const plainText = decoder(auth);
+
+  const requester = await User.getUserBy(plainText.email);
+
+  if (requester.permission === permission.ADMINISTRATOR) {
+    const user = req.body
+      ? {
+          firstName: req.body.firstName,
+          lastName: req.body.lastName,
+          email: req.body.email,
+          password: req.body.password
+        }
+      : {};
+
+    const signErrors = validateUser(user);
+
+    try {
+      if (!signErrors.valid) throw errors.signupError(signErrors.messages);
+
+      user.password = bcrypt.hashSync(user.password, salt);
+
+      const result = await User.getUserBy(user.email);
+      let valueMessage = '';
+      if (result) {
```

Comment:
This `if` is unneeded, because `createAdminUser` (name is a little tricky) create or update an user